nemo/utils/callbacks/nemo_model_checkpoint.py

```diff
@@ -240,11 +252,18 @@ def on_save_checkpoint(self, trainer, pl_module, checkpoint):
             if torch.distributed.is_initialized():
                 torch.distributed.barrier()
             backup_path = self._backup_existing_nemo_ckpt(trainer)
-            pl_module.save_to(save_path=app_state.model_restore_path)
-            logging.info(f"New .nemo model saved to: {app_state.model_restore_path}")
-        if backup_path is not None and is_global_rank_zero():
-            logging.info(f'Removing old .nemo backup {backup_path}')
-            get_filesystem(backup_path).rm(backup_path)
+            if not self.remove_backup_path:
+                consumed_samples = trainer.fit_loop.epoch_loop.batch_progress.total.processed
+                pl_module.save_to(
```

Comment:
Why the new model is saved or not based on `remove_backup_path` flag?

Reply:
cc @gshennvm